src/globus_cli/commands/gcs/endpoint/set_subscription_id.py

```diff
@@ -0,0 +1,72 @@
+from __future__ import annotations
+
+import typing as t
+import uuid
+
+import click
+
+from globus_cli.constants import EXPLICIT_NULL, ExplicitNullType
+from globus_cli.login_manager import LoginManager
+from globus_cli.parsing import command, endpoint_id_arg
+from globus_cli.termio import TextMode, display
+
+
+class GCSSubscriptionIdType(click.ParamType):
+    def get_type_annotation(self, _: click.Parameter) -> type:
+        return t.cast(type, uuid.UUID | t.Literal["DEFAULT"] | ExplicitNullType)
+
+    def convert(
+        self, value: str, param: click.Parameter | None, ctx: click.Context | None
+    ) -> t.Any:
+        if value is None or (ctx and ctx.resilient_parsing):
+            return None
```

Comment:
Can `value` be `None`? It isn't reflected in the function annotations.

Reply:
Looking at the super type, it cannot actually be None; will update in new rev.